### ISSUE TITLE
First weights aren't changed

### ISSUE THREAD

**Issue by Locust2520** [aymericdamien/TensorFlow-Examples#186]
Hey,

I came here to see how to build a simple _multilayer_ neural network with Tensorflow.
By searching on the Internet, I made a similar structure and trained it with the MNIST dataset.
The problem was that the weights of the first layer were not modified by Tensorflow.
And here, I see the same problem. By adding `print(sess.run(weights["h1"]))` before and after training, I note that no weights are changed:

`============== RESTART: F:/Python_docs/multilayer_perceptron.py ==============`
`Extracting /tmp/data/train-images-idx3-ubyte.gz`
`Extracting /tmp/data/train-labels-idx1-ubyte.gz`
`Extracting /tmp/data/t10k-images-idx3-ubyte.gz`
`Extracting /tmp/data/t10k-labels-idx1-ubyte.gz`
`[[ 0.2627787   0.77253848  0.72090644 ..., -0.8395927   0.68004286   1.30502164]`
` [ 0.99761492 -0.01503092  0.62924266 ...,  0.48425186 -0.19999036   -1.33137429]`
` [-0.54566431  1.3546375  -0.40144747 ...,  0.05340334  0.13847962  -0.19645269]`
` ..., `
` [-1.90896392 -1.62004948 -0.95858216 ...,  0.44503477  0.21179596  -0.49425131]`
` [ 1.10898376  1.64390612 -0.41823426 ...,  1.66942286  1.10209513  -0.4875792 ]`
` [ 0.34461403  0.63909751 -0.74047667 ...,  0.87805903  0.25059167   0.62332183]]`
`Epoch: 0001 cost=315.877097341`
`Epoch: 0002 cost=100.358125843`
`Epoch: 0003 cost=73.461984484`
`Epoch: 0004 cost=59.104319474`
`Epoch: 0005 cost=50.186071112`
`Epoch: 0006 cost=44.156094409`
`Epoch: 0007 cost=39.272400636`
`Epoch: 0008 cost=35.394302475`
`Epoch: 0009 cost=32.364734821`
`Epoch: 0010 cost=29.855774622`
`Epoch: 0011 cost=27.646177892`
`Epoch: 0012 cost=25.890833708`
`Epoch: 0013 cost=24.690152457`
`Epoch: 0014 cost=23.357179784`
`Epoch: 0015 cost=21.855909508`
`Optimization Finished!`
`[[ 0.2627787   0.77253848  0.72090644 ..., -0.8395927   0.68004286   1.30502164]`
 `[ 0.99761492 -0.01503092  0.62924266 ...,  0.48425186 -0.19999036  -1.33137429]`
 `[-0.54566431  1.3546375  -0.40144747 ...,  0.05340334  0.13847962  -0.19645269]`
 `..., `
 `[-1.90896392 -1.62004948 -0.95858216 ...,  0.44503477  0.21179596  -0.49425131]`
 `[ 1.10898376  1.64390612 -0.41823426 ...,  1.66942286  1.10209513  -0.4875792 ]`
 `[ 0.34461403  0.63909751 -0.74047667 ...,  0.87805903  0.25059167   0.62332183]]`
`Accuracy: 0.8897`

In this case it's more accurate to keep only one layer... but this is obviously not what we want !
Tensorflow seems to consider the input data is in the first layer and doesn't touch the first weights.

How can I do to make the whole NN train ?

**Comment by Locust2520** [comment]
Forget what I said:
weights are effectively corrected, but here we only see the beginning and the end of the matrix. This part isn't changed because in the MNIST dataset the start and the end of each input systematically equals to zero.
So this area has no influence on the result, this is why the value of weights doesn't matter and isn't changed.
